alfasim_sdk/status.py

```diff
@@ -1,16 +1,93 @@
+from enum import Enum
+
 import attr
 from attr import attrib
+from attr.validators import instance_of
+from attr.validators import optional
+from barril.units import Scalar
 
+from alfasim_sdk._validators import list_of_strings
 from alfasim_sdk._validators import non_empty_str
 
 
 @attr.s(kw_only=True)
 class ErrorMessage:
+    """
+    The model_name and the message informed is displayed on the status monitor of ALFAsim.
+    The execution of a new simulation will not be allowed until the error is solved.
+    """
+
     model_name: str = attrib(validator=non_empty_str)
     message: str = attrib(validator=non_empty_str)
 
 
 @attr.s(kw_only=True)
 class WarningMessage:
+    """
+    The model_name and the message informed is displayed on the status monitor of ALFAsim.
+    The execution of a new simulation will be kept normal.
+    """
+
     model_name: str = attrib(validator=non_empty_str)
     message: str = attrib(validator=non_empty_str)
+
+
+@attr.s(frozen=True)
+class PluginInfo:
+    name = attr.attrib(validator=non_empty_str)
+    enabled = attr.attrib(validator=instance_of(bool))
+    models = attr.attrib(validator=list_of_strings)
+
+
+@attr.s(frozen=True)
+class PipelineSegmentInfo:
```

Comment:
I am a little confused with this class. I mean, this class should have a parent information such as `Pipeline`, where we can identify where it ends. If there is no information about total length of the `Pipeline`, it is impossible to know where a `PipelineSegment` ends and what is its length.

Reply:
Commit be02ab45f6e7b045f5e1d9b127a97c9da9314ad9 adds a new field `PipelinInfo` thas has the `total_length` and a segments attribute ;)